otherlibs/dynlink/dynlink_common.ml

```diff
@@ -347,6 +347,7 @@ module Make (P : Dynlink_platform_intf.S) = struct
     init ();
     let filename = dll_filename filename in
     match P.load ~filename ~priv with
+    | exception (DT.Error (Cannot_open_dynamic_library _) as exn) -> raise exn
```

Comment:
I quick glance at the implementation of `P.load` (it has two implementations, in `byte/dynlink.ml` and in `native/dynlink.ml`) shows that it already guards the code with a catch-all and wraps exceptions with `Cannot_open_dynamic_library` as needed. If that's indeed the case, then we could simply get rid of the exception cases in this pattern matching, and write `let handle, units = P.load ... in ...`, right?